gecko-media/gecko/glue/ImageContainer.cpp

```diff
@@ -710,18 +710,13 @@ CopyPlane(uint8_t *aDst, const uint8_t *aSrc,
     // Fast path: planar input.
     memcpy(aDst, aSrc, aSize.height * aStride);
   } else {
-    int32_t height = aSize.height;
-    int32_t width = aSize.width;
-    for (int y = 0; y < height; ++y) {
-      const uint8_t *src = aSrc;
-      uint8_t *dst = aDst;
-      // Slow path
-      for (int x = 0; x < width; ++x) {
-        *dst++ = *src++;
-        src += aSkip;
+    for (int i = 0; i < aSize.height; i++) {
+      const uint8_t* rowSrc = aSrc + aStride * i;
+      for (int j = 0; j < aSize.width; j++) {
+        *aDst = *rowSrc;
+        aDst++;
```

Comment:
It seems the difference between the old code and your new code is that your code assumes that the stride of the destination is the same as the width. The caller in RecyclingPlanarYCbCrImage::CopyData() is assuming that the destination image has lines as wide as the stride. So I don't think what you're doing here is correct; it seems the old code is correct and the new code is wrong. Are you sure you need this change?

Reply:
OK, I built your branch on my Mac, and I see bad behaviour before your change, and good behaviour after. I still don't think ignoring the stride is a good idea. Maybe the cause of the problem is we are populating the stride fields incorrectly or somesuch?

Reply:
I reused this code btw https://searchfox.org/mozilla-central/source/gfx/layers/MacIOSurfaceHelpers.cpp#51